compose/ui/ui/src/uikitInstrumentedTest/kotlin/android/compose/ui/test/UIKitInstrumentedTest.kt

```diff
@@ -0,0 +1,207 @@
+                        val height = placeable?.height ?: constraints.minHeight
+
+                        layout(width, height) {
+                            placeable?.placeRelative(0, 0)
+                        }
+                    }
+                )
+            },
+            coroutineContext = coroutineContext
+        )
+
+        window.rootViewController = composeContainer
+        window.makeKeyAndVisible()
+        waitUntil { onDraw }
+        waitForIdle()
+    }
+
+    private fun keyboardUserInfo(height: Dp, animated: Boolean): Map<Any?, *> {
+        val keyboardHeight = screenSize.height - height
+        val keyboardDuration = if (animated) 0.25 else 0.0
+        return mapOf(
+            UIKeyboardAnimationDurationUserInfoKey to NSNumber(double = keyboardDuration),
+            UIKeyboardFrameEndUserInfoKey to NSValue.valueWithCGRect(
+                CGRectMake(
+                    x = 0.0,
+                    y = keyboardHeight.value.toDouble(),
+                    width = screenSize.width.value.toDouble(),
+                    height = keyboardHeight.value.toDouble()
+                )
+            ),
+            UIKeyboardAnimationCurveUserInfoKey to NSNumber(unsignedInteger = 0uL)
+        )
+    }
+
+    fun showKeyboard(height: Dp = 216.dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillShowNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(height, animated)
+            )
+        )
+        keyboardHeight = height
+        resizeKeyboard(height, animated)
+    }
+
+    fun resizeKeyboard(newHeight: Dp, animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(newHeight, animated)
+            )
+        )
+        keyboardHeight = newHeight
+    }
+
+    fun hideKeyboard(animated: Boolean = true) {
+        NSNotificationCenter.defaultCenter().postNotification(
+            NSNotification(
+                name = UIKeyboardWillChangeFrameNotification,
+                `object` = null,
+                userInfo = keyboardUserInfo(0.dp, animated)
+            )
+        )
+        keyboardHeight = 0.dp
+    }
+
+    private val isIdle: Boolean
+        get() {
+            val hadSnapshotChanges = Snapshot.current.hasPendingChanges()
+            val isApplyObserverNotificationPending = Snapshot.isApplyObserverNotificationPending
+            val containerInvalidations = composeContainer.hasInvalidations()
+
+            return !hadSnapshotChanges && !isApplyObserverNotificationPending && !containerInvalidations
+        }
+
+    fun waitForIdle(timeoutMillis: Long = 5_000) {
+        val defaultIdleScore = 1
```

Comment:
The code for dealing with idle score doesn't seem needed anymore.

Reply:
true, thanks